presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTester.java

```diff
@@ -52,6 +53,15 @@ public RuleAssert assertThat(Rule rule)
         return new RuleAssert(metadata, session, rule);
     }
 
+    public void assertInTransaction(Rule rule, Consumer<RuleAssert> ruleAssertion)
```

Comment:
Ah, that won't work as the tableScan node will not have enough information to match with pattern.